### DISCUSSION TITLE
need help about using ESP32-TWAI-CAN addon

### DISCUSSION THREAD

**Discussion by rcpilot007** [handmade0octopus/ESP32-TWAI-CAN#8] [General]
Hello. I'd like to make a "cruise control" (tempomat) for my Lada Kalina2 car (Itelma M74 CAN controller), where are 500 kbps CAN network inside. My idea is connecting my ESP32 WROOM 32/32E/D1Mini (ESP32 Dev module) onto the +12V-CANL-CANH-GND wires of MMC car media center or combined speedometer, to get some informations from ECU, maybe control speed of car, disable IMMO, etc., using only CAN network.

[mar.kesz.az.aramkor.ESP32WR32ECAN_TJA1050_MCP2515_test.pdf](https://github.com/user-attachments/files/16867488/mar.kesz.az.aramkor.ESP32WR32ECAN_TJA1050_MCP2515_test.pdf)
[Lada.Kalina2.CAN.sketch.sample.txt](https://github.com/user-attachments/files/16867502/Lada.Kalina2.CAN.sketch.sample.txt)

Now I tried to use Sandeep Mistry's routines; they are working, but I can't modify 0x12 ID address... 
I know, that ESP32 chips (I use ESP32-WROOM-32, 32E, D1 mini dev boards) can manages TWAI mode (which is compatible with CAN), I need to know better Your routines.

Main goal of this project, that I make a very small PCB (ESP32-D1 Mini is good for it) using TJA1050 CAN interface between ESP RX(16) TX(17) serial pins and CAN network of my car; using a very simple sketch -- to get some parameters from ECU and from speedometer, and send any others to the network; modified gas pedal state values, disabling IMMO, etc,.,

If it is possible, please share with me a very simple trasmit and receive sketch, which I can use -- like my sample sketch.

**Comment by handmade0octopus** [comment]
Simple send and receive sketch is in examples:
https://github.com/handmade0octopus/ESP32-TWAI-CAN/blob/master/examples/OBD2-querry/OBD2-querry.ino

**Comment by rcpilot007** [comment]
Thank You very much!

Maciej sorek Loboda ***@***.***> ezt írta (időpont: 2024.
szept. 4., Sze 21:26):

**Comment by CTBS-Terry** [comment]
I have implemented the library and it is working. My question is whether there is any provision to react to receipt of a CAN message and trigger an interrupt. It looks like this requires polling and I want to be able to be doing other tasks and get an interrupt that a message has been received.
